src/nvim/path.c

```diff
@@ -407,8 +407,9 @@ void add_pathsep(char *p)
 /// @param fname is the filename to save
 /// @param force is a flag to expand `fname` even if it looks absolute
 ///
-/// @return [allocated] Copy of absolute path to `fname` or NULL when
-///                     `fname` is NULL.
+/// @return [allocated]
+///   - Copy of absolute path to `fname` (truncated if longer than MAXPATHL)
+///   - NULL when `fname` is NULL or `vim_FullName()` fails
```

Comment:
We can't return NULL for failure, for reason mentioned in https://github.com/neovim/neovim/pull/5737#discussion_r91745980. Some call sites _expect_ the full, un-expanded path to be returned. 

We could examine all the calls to FullName_save(), but that could be a big refactor project, and would entail replacing `FullName_save()` with a new function so that we don't get confused when merging patches from Vim.